packages/node_modules/@ciscospark/react-container-activity-list/src/formatters/at-mention.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import {base64} from '@ciscospark/common';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './at-mention.css';
+
+const EVENT_NAME_MENTION_CLICKED = `mention:clicked`;
+
+function AtMentionComponent({
+  content,
+  onEvent
+}) {
+
+  function isMention(target) {
+    return target.tagName === `SPARK-MENTION` && target.dataset && typeof onEvent === `function`;
+  }
+
+  function handleClick(e) {
+    triggerEvent(e);
+  }
+
+  function handleKeyUp(e) {
+    if (e.keyCode && e.keyCode === 13) {
+      return triggerEvent(e);
+    }
+    return false;
+  }
+
+  function triggerEvent(e) {
+    const target = e.target;
+    if (isMention(target)) {
+      onEvent(EVENT_NAME_MENTION_CLICKED, {
+        type: target.dataset.objectType,
+        id: constructHydraId(target.dataset.objectType, target.dataset.objectId)
+      });
+    }
+  }
+
+  return ( // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+    <div
+      className={classNames(styles.atMention)}
+      dangerouslySetInnerHTML={{__html: content}} // eslint-disable-line react/no-danger
+      onClick={handleClick}
+      onKeyUp={handleKeyUp}
+      role="presentation"
+    />
+  );
+}
+
+AtMentionComponent.propTypes = {
+  content: PropTypes.string,
+  onEvent: PropTypes.func
+};
+
+export default (activity, onEvent) => {
+  if (activity.mentions) {
+    const component = ( // eslint-disable-line no-extra-parens
+      <AtMentionComponent
+        content={activity.content}
+        mentions={activity.mentions.items}
+        onEvent={onEvent}
+      />
+    );
+
+    return Object.assign({}, activity, {component});
+  }
+  return activity;
+};
+
+function constructHydraId(type, id) {
```

Comment:
We should eventually move this to a utility function, it is all over the codebase.

Reply:
Agreed. I wanted to but didn't want to create another package in this PR for it.